packages/@aws-cdk/toolkit-lib/test/api/cloud-assembly/source-builder.test.ts

```diff
@@ -114,6 +116,24 @@ describe('fromCdkApp', () => {
     expect(assembly.cloudAssembly.stacksRecursively.map(s => s.hierarchicalId)).toEqual(['Stack1', 'Stack2']);
   });
 
+  test('synth succeeds when working directory is given and outdir is relative', async () => {
+    // WHEN
+    const app = await appFixtureCommand('two-empty-stacks');
+    const cx = await toolkit.fromCdkApp(app.app, {
+      workingDirectory: app.workingDirectory,
+      outdir: 'relative.dir',
+    });
+
+    try {
+      await using assembly = await cx.produce();
+
+      // THEN - synth succeeds
+      expect(assembly.cloudAssembly.stacksRecursively.map(s => s.hierarchicalId)).toEqual(['Stack1', 'Stack2']);
+    } finally {
+      await fs.rm(path.join(app.workingDirectory, 'relative.dir'), { force: true, recursive: true });
```

Comment:
This should be automatic with `deleteOutdir`, right?

Reply:
Good point!